test/planner/plan_util_test.cpp

```diff
@@ -163,7 +164,139 @@ TEST_F(PlanUtilTests, GetAffectedIndexesTest) {
+      catalog::Column(type::TypeId::VARCHAR, 32, "first_name", false);
+  auto lname_column =
+      catalog::Column(type::TypeId::VARCHAR, 32, "last_name", false);
+
+  std::unique_ptr<catalog::Schema> table_schema(
+      new catalog::Schema({id_column, fname_column, lname_column}));
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  catalog->CreateTable(TEST_DB_COLUMNS, "test_table", std::move(table_schema),
+                       txn);
+  txn_manager.CommitTransaction(txn);
+
+  // Obtain ids for the table and columns
+  txn = txn_manager.BeginTransaction();
+  auto source_table = db->GetTableWithName("test_table");
+  oid_t table_id = source_table->GetOid();
+  oid_t id_col_oid =
+      source_table->GetSchema()->GetColumnID(id_column.column_name);
+  oid_t fname_col_oid =
+      source_table->GetSchema()->GetColumnID(fname_column.column_name);
+  oid_t lname_col_oid =
+      source_table->GetSchema()->GetColumnID(lname_column.column_name);
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  // This is required so that database objects are cached
+  auto db_object = catalog->GetDatabaseObject(TEST_DB_COLUMNS, txn);
+  EXPECT_EQ(1, static_cast<int>(db_object->GetTableObjects().size()));
+
+  auto table_object = db_object->GetTableObject("test_table");
+
+  // id and first_name are affected
+  std::string query_string = "UPDATE test_table SET id = 0, first_name = '';";
+  std::unique_ptr<Statement> stmt(new Statement("UPDATE", query_string));
+  auto &peloton_parser = parser::PostgresParser::GetInstance();
+  auto sql_stmt_list = peloton_parser.BuildParseTree(query_string);
+  std::vector<planner::col_triplet> affected_cols_vector =
+      planner::PlanUtil::GetAffectedColumns(
+          txn->catalog_cache, std::move(sql_stmt_list), TEST_DB_COLUMNS);
+  std::set<planner::col_triplet> affected_cols(affected_cols_vector.begin(),
+                                               affected_cols_vector.end());
+
+  EXPECT_EQ(2, static_cast<int>(affected_cols.size()));
+  std::set<planner::col_triplet> expected_oids;
+  expected_oids.emplace(database_id, table_id, id_col_oid);
+  expected_oids.emplace(database_id, table_id, fname_col_oid);
+  EXPECT_EQ(expected_oids, affected_cols);
+
+  // first_name is affected
+  query_string = "UPDATE test_table SET first_name = '';";
+  stmt.reset(new Statement("UPDATE", query_string));
+  sql_stmt_list = peloton_parser.BuildParseTree(query_string);
+  affected_cols_vector = planner::PlanUtil::GetAffectedColumns(
+      txn->catalog_cache, std::move(sql_stmt_list), TEST_DB_COLUMNS);
+  affected_cols = std::set<planner::col_triplet>(affected_cols_vector.begin(),
+                                                 affected_cols_vector.end());
+
+  EXPECT_EQ(1, static_cast<int>(affected_cols.size()));
+  expected_oids.clear();
+  expected_oids.emplace(database_id, table_id, fname_col_oid);
+  EXPECT_EQ(expected_oids, affected_cols);
```

Comment:
Add some predicates to the UPDATE statements. We should return the columns in those predicates.